pkg/healthcheck/healthcheck.go

```diff
@@ -330,7 +321,13 @@ func (hc *HealthChecker) addLinkerdVersionChecks() {
 			description: "data plane is up-to-date",
 			fatal:       false,
 			check: func() error {
-				return hc.kubeAPI.CheckProxyVersion(hc.dataPlanePods, hc.latestVersion)
+				for _, pod := range hc.dataPlanePods {
```

Comment:
Related to my comment above, as written `dataPlanePods` includes all pods, not just those pods that have been injected to point to `hc.ControlPlaneNamespace`. And uninjected pods will fail this check. I'd actually remove the `dataPlanePods` field altogether, in favor of the `getDataPlanePods` function that I suggested above. And then here you could just re-fetch the pods as needed. Something like:

```
			check: func() error {
				pods, err := hc.getDataPlanePods()
				if err != nil {
					return err
				}

				for _, pod := range pods {
					if pod.ProxyVersion != hc.latestVersion {
						return fmt.Errorf("%s is running version %s but the latest version is %s",
							pod.Name, pod.ProxyVersion, hc.latestVersion)
					}
				}

				return nil
			},
```

Reply:
I like that, I did not like that mutable field anyway but I thought that's an optimisation you agreed to do to prevent multiple requests to an API